client/src/cipher.rs

```
@@ -0,0 +1,165 @@
+    // create cipher instance
+    let mut cipher = AesOfb::new_var(&*key.0, &*key.1)?;
+
+    //Read file
+    let mut f = File::open(input_file)?;
+    let mut buffer = Vec::new();
+
+    // read the whole file
+    f.read_to_end(&mut buffer)?;
+
+    //stream ciphers
+    for chunk in buffer.chunks_mut(16) {
+        cipher.apply_keystream(chunk);
+    }
+
+    //Save file
+    let mut file = File::create(output_file)?;
+    file.write_all(&buffer)?;
+
+    Ok(())
+}
+
+fn ciphertext_path(plaintext_filename: &str) -> PathBuf {
+    let mut path = PathBuf::from(plaintext_filename);
+    path.set_extension("ciphertext");
+    path
+}
+
+fn decrypted_path(ciphertext_filename: &str) -> PathBuf {
+    let mut path = PathBuf::from(ciphertext_filename);
+    path.set_extension("decrypted");
+    path
+}
+
+fn keyfile_path(plaintext_filename: &str) -> PathBuf {
+    let mut path = PathBuf::from(plaintext_filename);
+    path.set_extension("aes256");
+    path
+}
+
+///Encrypt a file with AES256 and save it in the same folder as file_name.ciphertext
+///Create a key and save it into a file it in the same folder as file_name.aes256 if key is None
+pub fn encrypt(plaintext_filename: &str, key: Option<Key>) -> Result<()> {
+
+    let encryption_key = match key {
+        None => {
+            recover_or_generate_encryption_key(
+                &keyfile_path(plaintext_filename)
+            )?
+        }
+        Some(key) => key,
+    };
+
+    de_or_encrypt_file(
+        &Path::new(plaintext_filename), 
+        &ciphertext_path(plaintext_filename), 
+        encryption_key
+    )
+}
+
+///Decrypt a file with AES256 and save it in the same folder as file_name.decrypted
+///Create a key and save it into a file it in the same folder as file_name.aes256 if key is None
+pub fn decrypt(ciphertext_filename: &str, key: Option<Key>) -> Result<()> {
+
+    let encryption_key = match key {
+        None => {
+            recover_encryption_key(
```

@echevrier if you try to decrypt without keyfile this should fail early. makes no sense in generating a key if it doesn't exists (check commit b28b11bf22

---

Yes. this makes no sense